using-timescaledb/compression.md

```diff
@@ -19,184 +19,237 @@ basics of setting this up for use in your environment.
+`segmentby` list to the `orderby` list (described below).
 
-``` sql
-ALTER TABLE metrics 
-  SET (timescaledb.compress, 
-       timescaledb.compress_orderby = 'time',
-       timescaledb.compress_segmentby = 'device_id');
-```
 
-Using `timescaledb.compress_segmentby` takes the column you pass into it and
-creates a compressed array for each value in that column. The column itself is
-saved in uncompressed format, meaning it can be used in a b-tree index.
+## Understanding How Compression Works [](how-it-works)
+
+Our high-level approach to building columnar storage is to convert many wide rows of data (say, 1000) into a single row of data. But now, each field (column) of that new row stores an ordered set of data comprising the entire column of the 1000 rows.
+
+So, let’s consider a simplified example using a table that looks as follows:
+
+|time|device_id|cpu|disk_io|energy_consumption|
+|---|---|---|---|---|
+| 12:00:02 |1|88.2|20|0.8|
+| 12:00:01 |2|300.5|30| 0.9|
+| 12:00:01 |1|88.6|25|0.85|
+| 12:00:01 |2|299.1|40| 0.95|
+
+After converting this data to a single row, the data in “array” form:
+
+|time|device_id|cpu|disk_io|energy_consumption|
+|---|---|---|---|---|
+| [12:00:02, 12:00:02, 12:00:01, 12:00:1 ]| [1, 2, 1, 2]|[88.2, 300.5, 88.6, 299.1]|[20, 30, 25, 40] |[0.8, 0.9, 0.85, 0.95]| 
+
+#### Understanding the segmentby option [](understanding-segmentby)
+
+We can segment compressed rows by specific columns, so that each compressed
+row corresponds to data about a single item, e.g., a specific `device_id`.
+The `segmentby` option forces the system to break up the compressed array so
+that each compressed row has a single value for each segmentby column. For
+example if we set `device_id` as a `segmentby` column, then the compressed
+version of our running example would look like:
+
 
-So now your compressed data will look more like the following, which is divided
-into two segments (by `device_id`) and we can index `device_id` directly.
+|time|device_id|cpu|disk_io|energy_consumption|
+|---|---|---|---|---|
+| [12:00:02, 12:00:01]| 1 |[88.2, 88.6]|[20, 25] |[0.8, 0.85]| 
+| [12:00:02, 12:00:01]| 2 |[300.5, 299.1]|[30, 40] |[0.9, 0.95]| 
+ 
+The above example shows the the `device_id` column is no longer an array,
+instead it defines the single value associated with all of the compressed data in the row. 
 
-|device_id|time|value|
-|---|---|---|
-|1|8/22/2019 0:00|88.2|
-| |8/22/2019 1:00|88.6|
-| |8/22/2019 2:00|90.0|
-|2|8/22/2019 0:05|300.5|
-| |8/22/2019 1:05|302.0|
-| |8/22/2019 2:05|301.0|
+Because a single value is associated with a compressed row, no decompression
+is necessary to evaluate the value. Queries with WHERE clauses that filter by
+a `segmentby` column are much more efficient, as decompression can happen _after_
+filtering instead of before (thus avoiding the need to decompress
+filtered-out rows altogether). In fact, for even more efficient access,
+we build b-tree indexes over each `segmentby` column.
```

Comment:
Is this by default? Can you turn this off?

If so: "column (by default)."

Reply:
You cannot turn this off yet